### PR TITLE
compiler: Get rid of slops in structs

### PR DESCRIPTION
**Pull request by lpereira**
Regain space lost due to alignment by reordering structs from larger to smaller members and not intertwining bitfields with full-size fields.

With this change, the sizes are now:

    Struct       Size before    Size after
    assembler    104            96
    basicblock_  64             56

Sizes measured on x86-64 Linux, using [pahole](https://github.com/acmel/dwarves) and this [handy script](https://gist.github.com/mdboom/6052da76e152ccc9dc6b9741a7ddd6c6) that compares struct sizes between versions.

I don't think this warrants an issue, as it just rearranges members inside the `compile.c` translation unit.

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -233,19 +233,23 @@ typedef struct basicblock_ {
...
-    /* number of instructions used */
-    int b_iused;
-    /* length of instruction array (b_instr) */
-    int b_ialloc;
+    /* Exception stack at start of block, used by assembler to create the exception handling table */
+    ExceptStack *b_exceptstack;
     /* pointer to an array of instructions, initially NULL */
     struct instr *b_instr;
-    /* If b_next is non-NULL, it is a pointer to the next
+     /* If b_next is non-NULL, it is a pointer to the next
```
Comment (brandtbucher) [inline Python/compile.c]:
```suggestion
    /* If b_next is non-NULL, it is a pointer to the next
```

**Review by brandtbucher — APPROVED** [review]
Thanks! Found one minor whitespace issue, otherwise looks good:

File: Python/compile.c (hunk 1 above)
Comment (lpereira) [reply Python/compile.c]: Fixed, thank you.

**Comment by brandtbucher** [comment]
You can also add your name to `Misc/ACKS` too, if you want.

**Pull request merged**